libxml2.go

```diff
@@ -187,6 +187,54 @@ static struct xsdParserResult cParseUrlSchema(const char *url, const short int o
 	return parserResult;
 }
 
+static struct xsdParserResult cParseXsdMem(const void *xsd, const int goXsdSourceLen, const short int options) {
+	xmlLineNumbersDefault(1);
+	bool err = false;
+	struct xsdParserResult parserResult;
+	char *errBuf=NULL;
+	struct errCtx ectx;
+	ectx.errBuf=calloc(GO_ERR_INIT, sizeof(char));
+
+	xmlSchemaPtr schema = NULL;
+	xmlSchemaParserCtxtPtr schemaParserCtxt = NULL;
+
+	schemaParserCtxt = xmlSchemaNewMemParserCtxt(xsd, goXsdSourceLen);
+
+	if (schemaParserCtxt == NULL) {
```

Comment:
Pls. reuse common code parts, like this:
	
    static struct xsdParserResult parseSchema(xmlSchemaParserCtxtPtr schemaParserCtxt, const short int options) {
		bool err = false;
		struct xsdParserResult parserResult;
		char *errBuf=NULL;
		struct errCtx ectx;
		ectx.errBuf=calloc(GO_ERR_INIT, sizeof(char));
	
		xmlSchemaPtr schema = NULL;
	
		if (schemaParserCtxt == NULL) {
			err = true;
			strcpy(ectx.errBuf, "Xsd parser internal error");
		}
		else
		{
			if (options & P_ERR_VERBOSE) {
				xmlSchemaSetParserErrors(schemaParserCtxt, noOutputCallback, noOutputCallback, NULL);
				xmlSetGenericErrorFunc(&ectx, genErrorCallback);
			} else {
				xmlSetGenericErrorFunc(NULL, noOutputCallback);
				xmlSchemaSetParserErrors(schemaParserCtxt, genErrorCallback, noOutputCallback, &ectx);
			}
	
			schema = xmlSchemaParse(schemaParserCtxt);
	
			xmlSchemaFreeParserCtxt(schemaParserCtxt);
			if (schema == NULL) {
				err = true;
				char *tmp = malloc(strlen(ectx.errBuf) + 1);
				memcpy(tmp, ectx.errBuf, strlen(ectx.errBuf) + 1);
				free(ectx.errBuf);
				ectx.errBuf = tmp;
			}
		}
	
		errBuf=malloc(strlen(ectx.errBuf)+1);
		memcpy(errBuf,  ectx.errBuf, strlen(ectx.errBuf)+1);
		free(ectx.errBuf);
		parserResult.schemaPtr=schema;
		parserResult.errorStr=errBuf;
		errno = err ? -1 : 0;
		return parserResult;
	}
	
	static struct xsdParserResult cParseUrlSchema(const char *url, const short int options) {
		xmlLineNumbersDefault(1);
	
		xmlSchemaParserCtxtPtr schemaParserCtxt = NULL;
		schemaParserCtxt = xmlSchemaNewParserCtxt(url);
		return parseSchema(schemaParserCtxt, options);
	
	}
	
	static struct xsdParserResult cParseMemSchema(const void *xsd, const int goXsdSourceLen, const short int options) {
		xmlLineNumbersDefault(1);
	
		xmlSchemaParserCtxtPtr schemaParserCtxt = NULL;
		schemaParserCtxt = xmlSchemaNewMemParserCtxt(xsd, goXsdSourceLen);
	
		return parseSchema(schemaParserCtxt, options);
	}
	
```

Reply:
Sorry took even longer than expected. Embedding the xsd schema is a good idea btw.

Reply:
Copied and pasted your example as-is.